### ISSUE TITLE
The milliseconds inside the "set" is not being converted to seconds (required by redis)

### ISSUE THREAD

**Issue by iangregsondev** [node-cache-manager/node-cache-manager-ioredis-yet#66]
I have submitted a pull request.

https://github.com/node-cache-manager/node-cache-manager-ioredis-yet/pull/65

I hope you find everything in order.

The change was straightforward.

**Comment by wangkunmeng** [comment]
use cache-manager 5.*。

**Comment by iangregsondev** [comment]
I already am

```
        "cache-manager": "^5.1.3",
        "cache-manager-ioredis-yet": "^1.0.1",
```

cache-manager does accept "ms", this is fine and works..

the integration with  cache-manager-ioredis-yet should convert "ms" to seconds.

it does this in the "mset" but not in the "set".

as per pull request.

Or am I missing something ?

**Comment by iangregsondev** [comment]
cache-manager passes down "ms", but redis (ioredis or what ever implementation, on the set requires seconds), passes it down uncoverted..

so 1000ms from cache-manager ends up going to the redis as "1000" seconds 

if you look at the mset, its already doing this conversion BUT not in the "set"

**Comment by wangkunmeng** [comment]
@iangregsondev you are right.

**Comment by wangkunmeng** [comment]
```ts
import { Config, Store, Ttl } from 'cache-manager';
import Redis, { Cluster, ClusterNode, ClusterOptions, RedisOptions } from 'ioredis';

export interface RedisClusterConfig {
    nodes: ClusterNode[];
    options?: ClusterOptions;
}

export interface CacheConfig extends Config {
    isMillisecondTtl?: boolean;
}

export class RedisStore implements Store {
    constructor(private readonly redis: Redis | Cluster, private readonly options: CacheConfig = {}) {}

    static create(
        options: (RedisOptions | { clusterConfig: RedisClusterConfig }) & CacheConfig = { isMillisecondTtl: true }
    ) {
        const redis =
            'clusterConfig' in options
                ? new Redis.Cluster(options.clusterConfig.nodes, options.clusterConfig.options)
                : new Redis(options);

        return new RedisStore(redis, options);
    }

    get client() {
        return this.redis;
    }

    get isMillisecondTtl() {
        return this.options.isMillisecondTtl === true;
    }

    isCacheable(value: any): boolean {
        if (this.options.isCacheable) {
            return this.options.isCacheable(value);
        }

        return ![null, undefined].includes(value);
    }

    getVal(value: any) {
        return JSON.stringify(value) || '"undefined"';
    }

    async get<T>(key: string): Promise<T | undefined> {
        const val = await this.redis.get(key);

        if (val === undefined || val === null) {
            return undefined;
        }

        return JSON.parse(val) as T;
    }

    async set<T>(key: string, data: T, ttl?: Ttl): Promise<void> {
        if (!this.isCacheable(data)) {
            throw new Error(`"${data}" is not a cacheable value`);
        }

        ttl = ttl ?? this.options?.ttl;

        if (ttl) {
            if (this.isMillisecondTtl) {
                await this.redis.set(key, this.getVal(data), 'PX', ttl);
            } else {
                await this.redis.set(key, this.getVal(data), 'EX', ttl);
            }
        } else {
            await this.redis.set(key, this.getVal(data));
        }
    }

    async del(key: string): Promise<void> {
        await this.redis.del(key);
    }

    async reset(): Promise<void> {
        await this.redis.flushall();
    }

    async mset(args: Array<[string, unknown]>, ttl?: Ttl): Promise<void> {
        ttl = ttl ?? this.options?.ttl;

        if (ttl) {
            const multi = this.redis.multi();

            for (const [key, value] of args) {
                if (!this.isCacheable(value)) {
                    throw new Error(`"${this.getVal(value)}" is not a cacheable value`);
                }

                if (this.isMillisecondTtl) {
                    multi.set(key, this.getVal(value), 'PX', ttl);
                } else {
                    multi.set(key, this.getVal(value), 'EX', ttl);
                }
            }

            await multi.exec();
        } else {
            await this.redis.mset(
                args.flatMap(([key, value]) => {
                    if (!this.isCacheable(value)) {
                        throw new Error(`"${this.getVal(value)}" is not a cacheable value`);
                    }

                    return [key, this.getVal(value)] as [string, string];
                })
            );
        }
    }

    mget(...args: string[]): Promise<unknown[]> {
        return this.redis
            .mget(args)
            .then(x => x.map(x => (x === null || x === undefined ? undefined : (JSON.parse(x) as unknown))));
    }

    async mdel(...args: string[]): Promise<void> {
        if (args.length) {
            await this.redis.del(args);
        }
    }

    keys(pattern = '*'): Promise<string[]> {
        return this.redis.keys(pattern);
    }

    ttl(key: string): Promise<number> {
        return this.redis.ttl(key);
    }
}
```

**Comment by iangregsondev** [comment]
Nice, that looks pretty good.

Do you know when we can have it merged?

**Comment by botika** [comment]
Fixes v1.0.2
